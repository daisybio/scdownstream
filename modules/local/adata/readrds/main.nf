process ADATA_READRDS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/anndata2ri_bioconductor-singlecellexperiment_anndata_r-seurat:c4b75a61a89ec006':
        'community.wave.seqera.io/library/anndata2ri_bioconductor-singlecellexperiment_anndata_r-seurat:5fae42aabf7a1c5f' }"

    input:
    tuple val(meta), path(rds)

    output:
    tuple val(meta), path("*.h5ad"), emit: h5ad
    path("*.pkl")                  , emit: obsm, optional: true
    path "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    template 'readrds.py'
}
