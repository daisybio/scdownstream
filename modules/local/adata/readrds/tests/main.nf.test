nextflow_process {
    name 'Test Process ADATA_READRDS'
    script '../main.nf'
    process 'ADATA_READRDS'

    tag "modules"
    tag "modules/local"
    tag "adata/readrds"
    tag "adata"

    test("test_adata_readrds") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/scdownstream/samples/SAMN14430801_custom_emptydrops_filter_matrix.rds", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                {assert process.success},
                {assert snapshot(process.out.h5ad).match("adata_readrds_h5ad")},
                {assert snapshot(process.out.versions).match("adata_readrds_versions")}
            )
        }
    }

    test("test_adata_readrds - stub") {
        options '-stub'
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/scdownstream/samples/SAMN14430801_custom_emptydrops_filter_matrix.rds", checkIfExists: true),
                ]
                """
            }
        }
        then {
            assertAll(
                {assert process.success},
                {assert snapshot(process.out.h5ad).match("adata_readrds_h5ad_stub")},
                {assert snapshot(process.out.versions).match("adata_readrds_versions_stub")}
            )
        }
    }
}
