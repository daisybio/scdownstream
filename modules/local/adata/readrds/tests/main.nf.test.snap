{
    "adata_readrds_versions": {
        "content": [
            [
                "versions.yml:md5,ca41f493fa72083e3875b0618963984c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T14:53:13.997590657"
    },
    "adata_readrds_h5ad": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.h5ad:md5,8a310f6ac2557969edc163c9d3b8a050"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T14:53:12.430708745"
    }
}